packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.base.tsx

```diff
@@ -189,20 +194,36 @@ export class DetailsRowBase extends BaseComponent<IDetailsRowBaseProps, IDetails
     const showCheckbox = selectionMode !== SelectionMode.none && checkboxVisibility !== CheckboxVisibility.hidden;
     const ariaSelected = selectionMode === SelectionMode.none ? undefined : isSelected;
 
-    const classNames = getClassNames(styles, {
-      theme: theme!,
-      isSelected,
-      canSelect: !isContentUnselectable,
-      anySelected: isSelectionModal,
-      checkboxCellClassName,
-      droppingClassName,
-      className,
-      compact
-    });
+    this._classNames = {
+      ...this._classNames,
+      ...getClassNames(styles, {
+        theme: theme!,
+        isSelected,
+        canSelect: !isContentUnselectable,
+        anySelected: isSelectionModal,
+        checkboxCellClassName,
+        droppingClassName,
+        className,
+        compact
+      })
+    };
+
+    if (!this._rowClassNames) {
```

Comment:
This is not ideal. Since `DetailsRowFields` is not part of the public API, I'm going to explore refactoring these style calculations into the component itself to avoid the need for this.

The problem here is the `classNames` & `rowClassNames` duality causing referential integrity to fail.

Reply:
Committed https://github.com/OfficeDev/office-ui-fabric-react/pull/8799/commits/f133e9c9bb85493da5c38ba23a23ebee61379e4a as a compromise for now as the refactor to mergeStyles requires API deprecations and additional thought.